Reorder JSON to be more consistent

SPConrad opened the pull request:
Increase width of cells to be more like real dashboards
Consolidate `Job` and `Widget` select boxes as they must be the same
Add Name Label to cells, determinative logic to display name/title/missing
Fix numberOfColumns to be int instead of string

SPConrad committed:
Reorder JSON to be more consistent
Increase width of cells to be more like real dashboards
Consolidate `Job` and `Widget` select boxes as they must be the same
Add Name Label to cells, determinative logic to display name/title/missing
Fix numberOfColumns to be int instead of string

iamsellek commented on main.js:
```diff
@@ -362,12 +362,10 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
         const widgetWidth = widget.width;
         const widgetHeight = widget.height;
         let key = widgetCol + "_" + widgetRow;
-        // console.log("REGULAR KEY : " + key);
```
👍

iamsellek commented on main.js:
```diff
@@ -379,40 +377,52 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
+            const id = addCell();
+            makeHiddenCell(document.getElementById("widget-box-" + id));
+        } else {
+            if (jsonObject.config[widget.config].widgetTitle !== undefined) {
```
If there's any chance that `jsonObject.config[widget.config]` will not exist here, you'll probably want to add a check for that first and `&&` it with this current check.

SPConrad replied:
Yep

iamsellek commented on main.js:
```diff
@@ -379,40 +377,52 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
+            widgetConfigLabel.innerText = widgetConfig
+            selectedWidget.value = widgetWidget;
+            
+            if (widgetConfig) {
```
On line 417, you're setting widgetConfigLabel to widgetConfig and then you're checking for it's existence here. Should 417 move into this if statemnt?

SPConrad replied:
Yes

iamsellek commented on main.js:
```diff
@@ -474,15 +484,16 @@ function generateJSON() {
 
-        const configJSONName = selectedJob ? (selectedJob + "_" + itemID) : "";
+        const configJSONName = getItemConfigName(itemID);
+        //widgetNameLabel ? (widgetNameLabel + "_" + itemID) : "";
```
Should this still be here?

SPConrad replied:
Nope

iamsellek commented on main.js:
```diff
@@ -543,7 +554,9 @@ function parseJSONString(jsonString, column, row) {
 //Generates the HTML for a cell in the grid
 function generateItemDOMFragment(id) {
-    let jobsDropDownHtml = createDropDownHTML(jobsListName, jobsList, id, jobsPlaceholderMessage);
+    // TODO: variable-ify these strings
```
Does this TODO need to be completed before this PR goes in?

SPConrad replied:
Nope

iamsellek commented on main.js:
```diff
@@ -562,6 +576,11 @@ function generateItemDOMFragment(id) {
 }
 
+function createLabelHTML(type, itemId, placeholderMessage) {
+    let html = '<p class="cell-' + type + '-label" id="' + type + '-label-' + itemId + '"> ' + placeholderMessage + '</p>'
```
Can you use ticks here? (Top left of keyboard). They tend to make stuff like this a ton easier to read. [Check here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for more info.

iamsellek commented on main.js:
```diff
@@ -710,8 +729,10 @@ function configEditedListener(configJsonDisplayTextArea) {
 //Listener that handles when the use selects a new job or widget, and pre-fills the config textfield with a template based on the item selected
 function handleListChangeEvent(selectElement, id) {
 
+    /// all of the jobs have the same name as the widgets, just use the widget name 
```
👍

SPConrad committed:
Add check for jsonObject.config[widget.config] to be undefined before trying to access it

iamsellek commented on main.js:
```diff
@@ -379,40 +377,51 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
-                makeHiddenCell(document.getElementById(id));
-                // console.log("HIDDEN KEY : " + key);
+
+        if (widget == undefined || jsonObject.config[widget.config]) {
```
I think you want a not in front of jsonObject here.

SPConrad replied:
🤦‍♂️

SPConrad committed:
Check if widget is UNDEFINED, not if it exists
Add styling to file upload so file name is visible
Move use of widgetConfig to after widgetConfig check
